src/target/stm32f4.c

```diff
@@ -242,6 +245,47 @@ bool stm32f4_probe(target *t)
 	return false;
 }
 
+static bool gd32f4_attach(target *t)
+{
+	if (!cortexm_attach(t))
+		return false;
+
+	target_mem_map_free(t);
+	target_add_ram(t, 0x10000000, 0x10000); /* 64 k CCM Ram*/
+	target_add_ram(t, 0x20000000, 0x50000);     /* 320 k RAM */
+
+	/* TODO implement DBS mode */
+	const int split = 12;
+	/* Bank 1*/
+	stm32f4_add_flash(t, 0x8000000, 0x10000,  0x4000,  0, split); /* 4 16K */
+	stm32f4_add_flash(t, 0x8010000, 0x10000,  0x10000,  4, split); /* 1 64K */
+	stm32f4_add_flash(t, 0x8020000, 0xe0000,  0x20000,  5, split); /* 7 128K */
+
+	/* Bank 2 */
+	stm32f4_add_flash(t, 0x8100000, 0x10000,  0x4000,  16, split); /* 4 16K */
+	stm32f4_add_flash(t, 0x8110000, 0x10000,  0x10000,  20, split); /* 1 64K */
+	stm32f4_add_flash(t, 0x8120000, 0xe0000,  0x20000,  21, split); /* 7 128K */
+
+	/* Third MB composed of 4 256 KB sectors, and uses sector values 12-15 */
+	stm32f4_add_flash(t, 0x8200000, 0x100000,  0x40000,  12, split);
+	return true;
+}
+
+
+bool gd32f4_probe(target *t)
+{
+	switch (t->part_id) {
```

Comment:
It would be better if this was written as `if (t->part_id != ID_GD32F470)` with an early return false. If a switch-case is required later, we would structure it a bit differently to maintain the early return so the code is kept toward the left margin